cmake/modules/FindHostTools.cmake

```diff
@@ -2,45 +2,120 @@
 #
 # Copyright (c) 2022, Nordic Semiconductor ASA
 
-# FindHostTools module for locating a set of tools to use on the host for
-# Zephyr development.
-#
-# This module will lookup the following tools for Zephyr development:
-# +---------------------------------------------------------------+
-# | Tool               | Required |  Notes:                       |
-# +---------------------------------------------------------------+
-# | Generic C-compiler | Yes      |  Pre-processing of devicetree |
-# | Zephyr-sdk         |          |                               |
-# | gperf              |          |                               |
-# | openocd            |          |                               |
-# | bossac             |          |                               |
-# | imgtool            |          |                               |
-# +---------------------------------------------------------------+
-#
-# The module defines the following variables:
-#
-# 'CMAKE_C_COMPILER'
-# Path to C compiler.
-# Set to 'CMAKE_C_COMPILER-NOTFOUND' if no C compiler was found.
-#
-# 'GPERF'
-# Path to gperf.
-# Set to 'GPERF-NOTFOUND' if gperf was not found.
-#
-# 'OPENOCD'
-# Path to openocd.
-# Set to 'OPENOCD-NOTFOUND' if openocd was not found.
-#
-# 'BOSSAC'
-# Path to bossac.
-# Set to 'BOSSAC-NOTFOUND' if bossac was not found.
-#
-# 'IMGTOOL'
-# Path to imgtool.
-# Set to 'IMGTOOL-NOTFOUND' if imgtool was not found.
-#
-# 'HostTools_FOUND', 'HOSTTOOLS_FOUND'
-# True if all required host tools were found.
+#[=======================================================================[.rst:
+FindHostTools
+*************
+
+Find and configure host tools required for Zephyr development.
+
+This module locates and configures a set of tools needed for Zephyr development
+on the host system. It sets up the toolchain and finds various utilities used
+during the build process.
+
+Tools Located by this Module
+============================
+
+The module searches for the following tools:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 20 10 40
+
+   * - Tool
```

Comment:
this makes this very hard to look at from a text editor, would it support the
```
+------+------+
| blah | blah |
+------+------+
```
style of layout?

Reply:
Yeah sure, we can def encourage people to stick to less "intrusive" ReStructuredText constructs - agreed it's a bit disruptive. My hope is that if we adopt this kind of approach we'll prob end up rewriting/cleaning up quite a bit of the existing doc, so maybe that table in this current form will go away anyway :)

Reply:
fwiw I was primarily interested in feedback regarding the idea/approach rather than whether the table from the random module I took as an example was perfectly formatted or not :)

Reply:
I like it, having the documentation rendered in on the doc pages is a big improvement and it functions on search too +1 from me

Reply:
> I like it, having the documentation rendered in on the doc pages is a big improvement and it functions on search too +1 from me

Awesome! I hope you will be able to help come the time to actually do some cleanup / doc authoring as I won't necessarily know which modules should be documented in priority, which are maybe "internal", ... :)